base.py

```diff
@@ -38,9 +42,23 @@ def transform(self, X: Union[pd.DataFrame, np.ndarray]):
                 X[i : i + batch_size] for i in range(0, len(X), batch_size)
             )
 
-            results = Parallel(n_jobs=self.n_jobs)(
-                delayed(self._calculate_fingerprint)(X_sub) for X_sub in args
-            )
+            if self.verbose > 0:
+                total_batches = min(self.n_jobs, len(X))
```

Comment:
Shouldn't this be as follows?
```
total_batches = len(X) // joblib.effective_n_jobs(self.n_jobs)
```

Reply:
I think not (by the way `effective_n_jobs` is already calculated in `self.n_jobs`) - this would result in obtaining a batch size for data. `total_batches` is only used by `tqdm` to set the number of steps in the progress bar , so it should contain the actual number of processes that we use. Either we can use whole `self.n_jobs` or the length of the data is less than it and we use `len(X)`.

Reply:
This makes sense